### PR TITLE
react list to virtualize (lazy load) followers/following

### PR DESCRIPTION
**Pull request by chrisnojima**
Add react-list which lets us lazy load avatars for people with long lists of followers (coyne specifically)

### CODE REVIEW

**Comment by chrisnojima** [comment]
- [ ] https://github.com/keybase/js-vendor-desktop/pull/20

File: shared/profile/friendships.desktop.js [hunk 1]
```diff
@@ -1,28 +1,51 @@
...
   onClick?: (username: string) => void
 };
 
-const UserEntry = ({onClick, username, uid, followsYou, following, thumbnailUrl}: UserEntryProps) => (
-  <Box style={userEntryContainerStyle} onClick={() => { onClick && onClick(username, uid) }}>
-    <Avatar style={userEntryAvatarStyle} size={64} url={thumbnailUrl} followsYou={followsYou} following={following} />
-    <Text type='BodySmall' style={userEntryUsernameStyle(followsYou)}>{username}</Text>
-  </Box>
-)
+class UserEntry extends PureComponent<void, UserEntryProps, void> {
```
Comment (chrisnojima) [inline shared/profile/friendships.desktop.js]: speed this up a little bit

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
@keybase/react-hackers ok this can move forward but someone else needs to own it. don't forget the js-deps dependency needs to be merged also

**Comment by MarcoPolo** [comment]
I got this

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -130,19 +131,13 @@ export function triggerIdentify (uid: string = '', userAssertion: string = ''
...
-      dispatch(pendingIdentify(userAssertion || uid, false))
-    }, 60e3)
-
     const status = getState().config.status
     const myUID = status && status.user && status.user.uid
 
     // Don't identify ourself
     if (allowSelf || myUID !== uid) {
       dispatch({type: Constants.identifyStarted, payload: null})
+      console.log('aaaaaa1', uid, userAssertion)
```
Comment (MarcoPolo) [inline shared/actions/tracker.js]: ✂️

File: shared/common-adapters/user-proofs.desktop.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // @flow
-import React, {Component} from 'react'
+import React, {Component, PureComponent} from 'react'
```
Comment (MarcoPolo) [inline shared/common-adapters/user-proofs.desktop.js]: Do you still need component?

File: shared/actions/tracker.js [hunk 4]
```diff
@@ -77,94 +77,67 @@ export function registerTrackerIncomingRpcs (): TrackerActionCreator {
...
+
     // If we have a pending identify no point in firing off another one
-    if (getState().tracker.pendingIdentifies[username]) {
+    if (tracker.pendingIdentifies[username]) {
       console.log('Bailing on simultaneous getProfile', username)
       return
     }
 
+    const uid = tracker.trackers[username] && tracker.trackers[username].userInfo && tracker.trackers[username].userInfo.uid
+    const goodTill = tracker.cachedIdentifies[uid]
```
Comment (MarcoPolo) [inline shared/actions/tracker.js]: `uid && tracker.cachedIdentifies...`

File: shared/common-adapters/user-proofs.desktop.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/common-adapters/user-proofs.desktop.js]: yes

File: shared/profile/friendships.desktop.js [hunk 5]
```diff
@@ -35,34 +58,45 @@ const userEntryUsernameStyle = followsYou => ({
 })
 
 class Render extends Component<void, Props, void> {
```
Comment (MarcoPolo) [inline shared/profile/friendships.desktop.js]: Not sure, I couldn't find good docs on PureComponent so leaving it for now

**Comment by MarcoPolo** [comment]
I'm ignoring circle ci error because it's a known bug and this already passed the previous commit

**Pull request merged**